vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/pipeline_embedding_info.cc

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/ml/pipeline/pipeline_embedding_info.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/time/time.h"
+
+namespace ads {
+namespace ml {
+namespace pipeline {
+
+EmbeddingPipelineInfo::EmbeddingPipelineInfo() = default;
+
+EmbeddingPipelineInfo::EmbeddingPipelineInfo(
+    EmbeddingPipelineInfo&& info) noexcept = default;
+
+EmbeddingPipelineInfo& EmbeddingPipelineInfo::operator=(
+    EmbeddingPipelineInfo&& info) noexcept = default;
+
+EmbeddingPipelineInfo::~EmbeddingPipelineInfo() = default;
+
+bool EmbeddingPipelineInfo::FromValue(base::Value value) {
+  base::Value::Dict* resource = value.GetIfDict();
+  if (!resource) {
+    return false;
+  }
+
+  absl::optional<int> version_value = resource->FindInt("version");
+  if (!version_value.has_value()) {
+    return false;
+  }
+  version = version_value.value();
+
+  std::string* timestamp_value = resource->FindString("timestamp");
+  if (!timestamp_value) {
+    return false;
+  }
+  base::Time timestamp;
+  bool success =
+      base::Time::FromUTCString((*timestamp_value).c_str(), &timestamp);
```

Comment:
Please move base::Time::FromUTCString inside the conditional statement, as then we do not need bool success